sdv/utils.py

```diff
@@ -31,9 +42,14 @@ def get_datetime_format(value):
         String representing the datetime format in ``strftime`` format or ``None`` if not detected.
     """
     if isinstance(value, pd.Series):
-        value = value.astype(str).to_list()
-    if not isinstance(value, (list, np.ndarray)):
-        value = [value]
+        value = value.astype(str).to_numpy()
+    elif isinstance(value, list):
+        value = np.array(value)
+    if not isinstance(value, np.ndarray):
+        value = np.array([value])
+
+    #  Fixes a bug that occurs in pandas as numpy casts to np.str_
+    value = np.array([get_fisrt_valid_value(value)], dtype=object)
```

Comment:
If we're only going to use the first value, why go through the trouble of converting all the lists/series above?